tests/app/views/test_suppliers.py

```diff
@@ -1749,3 +1753,261 @@ def test_changing_on_framework_to_passed_creates_audit_event(self, open_dos_fram
+            assert audit_events[0].data["supplierId"] == 1
+            assert audit_events[0].data["frameworkSlug"] == "g-cloud-8"
+            assert audit_events[0].data["variationSlug"] == "banana"
+            assert audit_events[0].data["update"] == {
+                "agreedUserId": 1,
+            }
+
+    def test_agree_variation_playpen(
+            self,
+            app,
+            live_g8_framework_2_variations_suppliers_on_framework_with_alt,
+            ):
+        with freeze_time('2016-06-06'):
+            response = self.client.put("/suppliers/2/frameworks/g-cloud-8/variation/toblerone", data=json.dumps({
+                "updated_by": "test123",
+                "agreedVariation": {
+                    "agreedUserId": 2,
+                },
+            }), content_type="application/json")
+
+        expected_variation_toblerone = {
+            "agreedUserId": 2,
+            "agreedUserEmail": "test+2@digital.gov.uk",
+            "agreedUserName": "my name",
+            "agreedAt": "2016-06-06T00:00:00.000000Z",
+        }
+
+        assert response.status_code == 200
+        response_json = json.loads(response.get_data())
+        assert response_json["agreedVariation"] == expected_variation_toblerone
+
+        response2 = self.client.get("/suppliers/2/frameworks/g-cloud-8")
+        assert response2.status_code == 200
+        assert json.loads(response2.get_data())["frameworkInterest"]["agreedVariations"] == {
+            "toblerone": expected_variation_toblerone,
+        }
+
+        # check we've left the other supplier alone
+        response3 = self.client.get("/suppliers/1/frameworks/g-cloud-8")
+        assert response3.status_code == 200
+        assert not json.loads(response3.get_data())["frameworkInterest"]["agreedVariations"]
+
+        with freeze_time("2016-07-07"):
+            response4 = self.client.put("/suppliers/2/frameworks/g-cloud-8/variation/banana", data=json.dumps({
+                "updated_by": "test123",
+                "agreedVariation": {
+                    "agreedUserId": 2,
+                },
+            }), content_type="application/json")
+
+        expected_variation_banana = {
+            "agreedUserId": 2,
+            "agreedUserEmail": "test+2@digital.gov.uk",
+            "agreedUserName": "my name",
+            "agreedAt": "2016-07-07T00:00:00.000000Z",
+        }
+
+        assert response4.status_code == 200
+        response4_json = json.loads(response4.get_data())
+        assert response4_json["agreedVariation"] == expected_variation_banana
+
+        response5 = self.client.get("/suppliers/2/frameworks/g-cloud-8")
+        assert response5.status_code == 200
+        assert json.loads(response5.get_data())["frameworkInterest"]["agreedVariations"] == {
+            "banana": expected_variation_banana,
+            "toblerone": expected_variation_toblerone,
+        }
+
+        # check we can't agree a second time
```

Comment:
It might be good to split this test out into a separate one as it seems like it's testing a separate bit of functionality